### DISCUSSION TITLE
Build keeps running out of memory

### DISCUSSION THREAD

**Discussion by isaacfink** [railwayapp/nixpacks#1146] [Q&A]
I keep getting an error code 137 during build of a simple sveltekit application, I can run the build command locally just fine but when using nixpacks it keeps throwing this error, here are some logs from the last run
```text
#13 96.72 127:     {/if}
#13 96.72 128:   </output>
#13 146.7 Killed
#13 146.7  ELIFECYCLE  Command failed with exit code 137.
#13 146.9  ELIFECYCLE  Command failed with exit code 137.
------
executor failed running [/bin/bash -ol pipefail -c pnpm run build:client]: exit code: 137
Error: Docker build failed
```
From what i understand nixpacks builds a docker image so my question is how can I allocate more ram to the docker images? I think my system is set to 2GB which should be enough for a simple build

**Comment by alessio-zamparelli** [comment]
maybe found a way:
pass NODE_OPTIONS=--max_old_space_size=4096 (or set the env in any way) as env for the build phase

**Comment by tamish-max** [comment]
Exit code `137` is the important clue here — that’s almost always the build being killed for running out of memory.

For Node/SvelteKit builds, the first thing I’d try is raising the Node heap during the build rather than trying to change Docker globally:

```toml
[variables]
NODE_OPTIONS = "--max-old-space-size=3072"
```

or, if you only want it for the build step:

```toml
[phases.build]
cmds = [
  "NODE_OPTIONS=--max-old-space-size=3072 pnpm run build:client"
]
```

A couple of other things that help with pnpm/SvelteKit builds:

- disable large sourcemaps in production builds if you don’t need them
- avoid running multiple heavy build commands in parallel
- if you’re in a monorepo, build only the target app instead of the entire workspace

If you’re using the Nixpacks CLI locally, you can also give the builder more memory directly. But on a hosted platform, the practical fix is usually “reduce build memory pressure or raise Node heap”, because the platform build container limit is what kills the process.
